### PR TITLE
[MJAVADOC-625] Support for multiple stylesheets

### PR DESCRIPTION
**Pull request by schelldorfer**

### CODE REVIEW

**Comment by michael-o** [comment]
I am confused. [This](https://bugs.openjdk.java.net/browse/JDK-8189217) says it can be called multiples times, yet you have defined it as string?

**Comment by michael-o** [comment]
Then I don't see the purpose of this incomplete PR.

**Comment by schelldorfer** [comment]
The main idea of this PR is to have an additional stylesheet without loosing the default stylesheet.

**Comment by michael-o** [comment]
You won't lose the default one. The JIRA issue says that this adds additional ones to the main stylesheet. Did I misread the JIRA issue?

**Comment by schelldorfer** [comment]
When using "stylesheet" / "stylesheetfile", this overwrites the default javadoc stylesheet. There is no possibility with the current Maven Javadoc plugin to keep the default javadoc stylesheet and have an additional custom stylesheet.
With this PR, this is possible (keep the default javadoc stylesheet and provide an additional custom stylesheet).

**Comment by michael-o** [comment]
I understand that, but if the original option can be called multiple times, it feels natural that this plugin supports this too.

**Comment by schelldorfer** [comment]
I've updated schelldorfer:MJAVADOC-625 branch to support **multiple** additional stylesheets

File: src/it/projects/MJAVADOC-625/pom.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>org.apache.maven.plugins.javadoc.it</groupId>
+	<artifactId>mjavadoc-625</artifactId>
+	<version>1.0.0-SNAPSHOT</version>
+
+	<url>https://issues.apache.org/jira/browse/MJAVADOC-625</url>
+
+	<build>
```
Comment (elharo) [inline src/it/projects/MJAVADOC-625/pom.xml]: Maven project code style is 2 space indents in XML: https://maven.apache.org/developers/conventions/code.html

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java [hunk 2]
```diff
@@ -1422,8 +1422,28 @@
...
-   private String stylesheetfile;
-
+    private String stylesheetfile;
+    
+    /**
+     * Specifies the path of an additional HTML stylesheet file.
+     * @since 3.1.2
+     */
+    @Parameter
+    private String addStylesheet;
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java]: additionalStylesheet

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java [hunk 3]
```diff
@@ -1422,8 +1422,28 @@
...
+    
+    /**
+     * Specifies the path of an additional HTML stylesheet file.
+     * Example:
+     * <pre>
+     *     &lt;addStylesheets&gt;
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java]: are tag names like this typically nouns or verbs?

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java [hunk 4]
```diff
@@ -2981,6 +3001,35 @@ private String getStylesheetFile( final File javadocOutputDirectory )
 
         return getResource( new File( javadocOutputDirectory, DEFAULT_CSS_NAME ), stylesheetfile );
     }
+    
+    private String getAddStylesheet( final File javadocOutputDirectory, final String stylesheet )
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java]: getAdditionalStylesheet

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java [hunk 5]
```diff
@@ -2981,6 +3001,35 @@ private String getStylesheetFile( final File javadocOutputDirectory )
...
+    
+    private String getAddStylesheet( final File javadocOutputDirectory, final String stylesheet )
+            throws MavenReportException
+    {
+        if ( StringUtils.isEmpty( stylesheet ) )
+        {
+            return null;
+        }
+        
+        File addstylesheetfile = new File( stylesheet );
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java]: additionalStyleheetFile

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java [hunk 6]
```diff
@@ -2981,6 +3001,35 @@ private String getStylesheetFile( final File javadocOutputDirectory )
...
+        if ( addstylesheetfile.exists() )
+        {
+            String stylesheetfilename = getStylesheetFile( javadocOutputDirectory );
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java]: stylesheetFilename

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java [hunk 7]
```diff
@@ -5432,6 +5481,17 @@ private void addStandardDocletOptions( File javadocOutputDirectory,
 
         addArgIfNotEmpty( arguments, "-stylesheetfile",
                           JavadocUtil.quotedPathArgument( getStylesheetFile( javadocOutputDirectory ) ) );
+        
+        addArgIfNotEmpty( arguments, "--add-stylesheet",
+                          JavadocUtil.quotedPathArgument( getAddStylesheet( javadocOutputDirectory, addStylesheet ) ) );
+        if ( addStylesheets != null && addStylesheets.length != 0 )
+        {
+            for ( String as : addStylesheets )
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java]: as --> stylesheet

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java (hunk 2 above)
Comment (rfscholte) [reply src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:1432]: the name of the parameter is `--add-stylesheet`, the parameter should reflect that.

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java (hunk 3 above)
Comment (rfscholte) [reply src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:1438]: if we would control the name of the parameter, most likely yet. But this one matches the javadoc tool parameter

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java (hunk 7 above)
Comment (rfscholte) [reply src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:5489]: agree

**Comment by rfscholte** [comment]
Merged with https://github.com/apache/maven-javadoc-plugin/commit/970c8a661fa284d1d9b5d3302660bf4f1e4e13e5

**Pull request closed without merging**